lib/tapjoy/ldap/key.rb

```diff
@@ -1,196 +1,96 @@
 
-          opt :user, 'Specify username to add key to', :type => :string,
-              :required => true
-          opt :filename, 'File to load keys from', :type => :string, :default => '-'
+        def remove
+          key = Tapjoy::LDAP::Key::Remove.new
+          key.remove
         end
 
-        keys = get_keys_from_commandline(opts[:filename])
-
-        filter = Net::LDAP::Filter.eq('uid', opts[:user])
-        results = Tapjoy::LDAP::client.search(attributes = ['*'], filter = filter)
-
-        # Make sure we return one, and only one user DN
-        if results.size < 1
-          abort 'user not found'
-        elsif results.size > 1
-          abort 'Multiple users found. Please narrow your search.'
+        def install
+          key = Tapjoy::LDAP::Key::Install.new
+          key.install
         end
 
-        results.each  do |result|
-          unless result.objectclass.include?('ldapPublicKey')
-            puts 'LDAP Public Key Object Class not found.'
-            abort 'Please ensure user was created correctly.'
-          end
-          keys.each do |key|
-            Tapjoy::LDAP::client.conn.add_attribute(result.dn, :sshPublicKey, key)
-            puts Tapjoy::LDAP::client.return_result
-          end
+        def list
+          Tapjoy::LDAP::Key.get_keys_from_ldap
         end
-      end
 
-      # Remove key from LDAP
-      def remove
-        opts = Trollop::options do
-          # Set help message
-          banner("#{$0} key remove [options]")
-
-          opt :user, 'Specify username to delete key from', :type => :string,
-              :required => true
-          opt :filename, 'File to load key deletion list from', :type => :string,
-              :default => '-'
-          opt(:force, 'Force delete')
+        def show
+          key = Tapjoy::LDAP::Key::Show.new
+          key.show
         end
 
-        keys = get_keys_from_commandline(opts[:filename])
-
-        filter = Net::LDAP::Filter.eq('uid', opts[:user])
-        attributes = ['sshPublicKey']
-        old_array = []
-
-        new_array = []
+        def get_keys_from_ldap
 
-        results = Tapjoy::LDAP::client.search(attributes, filter)
-        if results.size < 1
-          puts "User (#{ opts[:user] }) not found."
-          abort 'Please check the username and try again'
-        elsif results.size > 1
-          abort 'Multiple users found. Please narrow your search.'
+          key_results = {}
+          results = Tapjoy::LDAP::client.search(attributes = ['uid', 'sshPublicKey'],
+                      filter = Net::LDAP::Filter.eq('sshPublicKey', '*'))
+          results.each {|result| key_results[result.uid[0]] = result.sshPublicKey}
+          key_results
         end
 
-        results.each do |result|
-          @user_dn = result.dn
-          puts "User DN: #{ @user_dn }"
-          old_array = result.sshPublicKey
-        end
+        # Retrieve keys from file/stdin
+        def get_keys_from_commandline(filename=nil)
```

Comment:
I we're allowing the filename to be blank, we'll need to ask the end user what key they want revoked.